Framework/Algorithms/src/CalcCountRate.cpp

```diff
@@ -0,0 +1,653 @@
+#include "MantidAlgorithms/CalcCountRate.h"
+
+#include "MantidKernel/PropertyWithValue.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/TimeSeriesProperty.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidKernel/make_unique.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/NumericAxis.h"
+#include "MantidAPI/WorkspaceFactory.h"
+
+#include "MantidDataObjects/Workspace2D.h"
+#include <numeric>
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CalcCountRate)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CalcCountRate::name() const { return "CalcCountRate"; }
```

Comment:
Indeed, now when I think about it, `CountRateCalculation` doesn't fit the naming scheme of Mantid. However, I wouldn't shorten Calculate to Calc, even if there are too many algorithms starting with that prefix. It would just create another 'subcategory' of algorithms, similarly to the ComputeSomething algorithms: I do remember there is a vanadium calibration algorithm, but I don't remember whether it was CalculateCalibrationCoefVan or ComputeCalibrationCoefVan. Adding Calc as a possible prefix would just complicate things. So I guess renaming this algorithm back to `CalculateCountRate` would be the best choice.